packages/scene-composer/src/utils/entityModelUtils/modelRefComponent.ts

```diff
@@ -0,0 +1,97 @@
+import { ComponentRequest, ComponentUpdateRequest } from '@aws-sdk/client-iottwinmaker';
+import { DocumentType } from '@aws-sdk/types';
+
+import { IModelRefComponent, KnownComponentType } from '../../interfaces';
+import { componentTypeToId } from '../../common/entityModelConstants';
+import { IModelRefComponentInternal } from '../../store';
+import { generateUUID } from '../mathUtils';
+
+enum ModelRefComponentProperty {
+  Uri = 'uri',
+  ModelType = 'modelType',
+  UnitOfMeasure = 'unitOfMeasure',
+  LocalScale = 'localScale',
+  CastShadow = 'castShadow',
+  ReceiveShadow = 'receiveShadow',
+}
+
+export const createModelRefComponent = (model: IModelRefComponent): ComponentRequest => {
+  const comp: ComponentRequest = {
+    componentTypeId: componentTypeToId[KnownComponentType.ModelRef],
+    properties: {
+      [ModelRefComponentProperty.Uri]: {
+        value: {
+          stringValue: model.uri,
+        },
+      },
+      [ModelRefComponentProperty.ModelType]: {
+        value: {
+          stringValue: model.modelType,
+        },
+      },
+    },
+  };
+
+  if (model.unitOfMeasure) {
+    comp.properties![ModelRefComponentProperty.UnitOfMeasure] = {
+      value: {
+        stringValue: model.unitOfMeasure,
+      },
+    };
+  }
+  if (model.localScale) {
+    comp.properties![ModelRefComponentProperty.LocalScale] = {
+      value: {
+        listValue: model.localScale.map((l) => ({ doubleValue: l })),
+      },
+    };
+  }
+  if (model.castShadow !== undefined) {
+    comp.properties![ModelRefComponentProperty.CastShadow] = {
+      value: {
+        booleanValue: model.castShadow,
+      },
+    };
+  }
+  if (model.receiveShadow !== undefined) {
+    comp.properties![ModelRefComponentProperty.ReceiveShadow] = {
+      value: {
+        booleanValue: model.receiveShadow,
+      },
+    };
+  }
+  return comp;
+};
+
+export const updateModelRefComponent = (model: IModelRefComponent): ComponentUpdateRequest => {
+  const request = createModelRefComponent(model);
+  return {
+    componentTypeId: request.componentTypeId,
+    propertyUpdates: request.properties,
+  };
+};
+
+export const parseModelRefComp = (model: DocumentType): IModelRefComponentInternal | undefined => {
+  const uriExists = !model?.['properties'].some((prop) => prop['propertyName'] === 'uri');
+  const modelTypeExists = !model?.['properties'].some((prop) => prop['propertyName'] === 'modelType');
```

Comment:
the variable names are misleading. they really mean `xxxMissing` instead of `xxxExists`. they cause problem for people to understand why `uriExists || modelTypeExists` needs to return undefined

Reply:
also you can use `find` instead of `some` here. so when their value exist, you can just use those values to construct the component instead of `find` again below

Reply:
Actually if we want to reuse the value then we don't have to use missing or exists. We can just findUri/FindModelType. If undefined then return undefined or elase using `findUri.propertyValue`